src/exp/list_exp.rs

```diff
@@ -0,0 +1,752 @@
+// Copyright 2015-2020 Aerospike, Inc.
+//
+// Portions may be licensed to Aerospike, Inc. under one or more contributor
+// license agreements.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+//! List Cdt Aerospike Filter Expressions.
+
+use crate::exp::{ExpOp, ExpType, Expression, ExpressionArgument, FilterExpression, MODIFY};
+use crate::operations::cdt_context::CdtContext;
+use crate::operations::lists::{ListPolicy, ListReturnType, ListSortFlags};
+use crate::Value;
+
+/// List expression generator.
+///
+/// The bin expression argument in these methods can be a reference to a bin or the
+/// result of another expression. Expressions that modify bin values are only used
+/// for temporary expression evaluation and are not permanently applied to the bin.
+///
+/// List modify expressions return the bin's value. This value will be a list except
+/// when the list is nested within a map. In that case, a map is returned for the
+/// list modify expression.
+///
+/// List expressions support negative indexing. If the index is negative, the
+/// resolved index starts backwards from end of list. If an index is out of bounds,
+/// a parameter error will be returned. If a range is partially out of bounds, the
+/// valid part of the range will be returned. Index/Range examples:
+///
+/// * Index 0: First item in list.
+/// * Index 4: Fifth item in list.
+/// * Index -1: Last item in list.
+/// * Index -3: Third to last item in list.
+/// * Index 1 Count 2: Second and third items in list.
+/// * Index -3 Count 3: Last three items in list.
+/// * Index -5 Count 4: Range between fifth to last item to second to last item inclusive.
+///
+/// Nested expressions are supported by optional CTX context arguments.
+pub struct ListExpression {}
+
+const MODULE: i64 = 0;
+
+#[doc(hidden)]
+pub enum ListExpOp {
+    APPEND = 1,
+    AppendItems = 2,
```

Comment:
Why are some of these enums in UPPERCASE and some in TitleCase?

Reply:
Oh, i didnt really notice yet. I guess CLion did that automaticly because Enum format should be TitleCase. I originally copied them from the Java code. Seems like its only doing that for multi word keys. Ill fix that.